applications/luci-app-https_dns_proxy/luasrc/controller/https_dns_proxy.lua

```diff
@@ -1,7 +1,103 @@
 module("luci.controller.https_dns_proxy", package.seeall)
 function index()
-	if not nixio.fs.access("/etc/config/https_dns_proxy") then
-		return
+	if nixio.fs.access("/etc/config/https_dns_proxy") then
+		entry({"admin", "services", "https_dns_proxy"}, cbi("https_dns_proxy"), _("HTTPS DNS Proxy"))
+		entry({"admin", "services", "https_dns_proxy", "action"}, call("https_dns_proxy_action"), nil).leaf = true
 	end
-	entry({"admin", "services", "https_dns_proxy"}, cbi("https_dns_proxy"), _("HTTPS DNS Proxy"))
+end
+
+function https_dns_proxy_action(name)
+	local packageName = "https_dns_proxy"
+	if name == "start" then
+		servers_backup()
+		luci.sys.init.start(packageName)
+		servers_apply_doh()
+		luci.util.exec("/etc/init.d/dnsmasq restart >/dev/null 2>&1")
+	elseif name == "action" then
+		luci.util.exec("/etc/init.d/" .. packageName .. " reload >/dev/null 2>&1")
+		luci.util.exec("/etc/init.d/dnsmasq restart >/dev/null 2>&1")
+	elseif name == "stop" then
+		servers_restore()
+		luci.util.exec("/etc/init.d/dnsmasq restart >/dev/null 2>&1")
+		luci.sys.init.stop(packageName)
+	elseif name == "enable" then
+		luci.sys.init.enable(packageName)
+	elseif name == "disable" then
+		luci.sys.init.disable(packageName)
+	end
+	luci.http.prepare_content("text/plain")
+	luci.http.write("0")
+end
+
+local uci = require("luci.model.uci").cursor()
+
+function uci_del_list(conf, sect, opt, value)
+  local lval = uci:get(conf, sect, opt)
```

Comment:
Please fix the indentation

Reply:
I'm not sure I understand. The code of the function is indented, should it not be? Are you referring to something else?

Reply:
The indentation there is <space> but we prefer <tabs>

Reply:
Ah, I see. I'll play with the tab setting of my editor in the future, I still think this should be merged as is for now.

Reply:
I would like to do, but if we fix the indentation afterwards, then we lose the history if we want to find out with git blame when and what has been changed. This makes it more difficult to find errors when something was added and why. I'd rather see it if it is entered correctly.

Reply:
I believe I've been using the same editor for a while, so I believe double-spaces are in all of my packages, both luci and regular. What would be the best way to convert it to tabs now? Should I just run sed -i 's/  /	/g' on all files?

Reply:
I am using the following *.vimrc*. This helps me to see the tab and the whitespace in my code and also the boundary for 80 chars.
```
" Show whitespace char
set listchars=tab:\|.,trail:.,extends:>,precedes:<
set list

" Set char highlight columne
set colorcolumn=80
```
If you like then you should convert it to tabs but this is not necessary because it was already accepted.